setup.py

```diff
@@ -246,11 +251,28 @@ def finalize_options(self):
     def run(self):
         build_cmd = self.get_finalized_command('build')
         build_cmd.run()
-        sys.path.insert(0, build_cmd.build_lib)
-        sys.path.insert(0, 'tests/lib')
-        import test_all
-        if not test_all.main([]):
-            raise DistutilsError("Tests failed")
+
+        # running the tests this way can pollute the post-MANIFEST build sources
+        # (see https://github.com/yaml/pyyaml/issues/527#issuecomment-921058344)
+        # until we remove the test command, run tests from an ephemeral copy of the intermediate build sources
+        tempdir = tempfile.TemporaryDirectory(prefix='test_pyyaml')
```

Comment:
Can we put this in a `with`?

Reply:
Ultimately this whole "test from setup.py" thing needs to just go away and be replaced by `pytest`, but I didn't get around to dusting off my custom pytest discovery wrapper for pyyaml's "special" tests, and hiding the entire thing inside pytest as a single opaque test blob is worse.